crates/ra_syntax/src/grammar/items/traits.rs

```diff
@@ -115,3 +115,17 @@ fn choose_type_params_over_qpath(p: &Parser) -> bool {
     (p.nth(1) == LIFETIME || p.nth(1) == IDENT)
         && (p.nth(2) == R_ANGLE || p.nth(2) == COMMA || p.nth(2) == COLON || p.nth(2) == EQ)
 }
+
+// impl Type {}
+//      ^^^^
+// impl Trait for T {}
+//      ^^^^^
+pub(crate) fn impl_type(p: &mut Parser) {
```

Comment:
You can turn these comments into executable tests, by changing them to 

```
// test impl_type
// impl Trait for Type {}
// impl impl NotType {}
// impl Trait for impl NotType {}
```

, running `cargo gen-tests` to generate tests at tests/data/parser/inline and running `cargo test` to generate the gold answer.